examples/components/http-hello-world/README.md

```diff
@@ -0,0 +1,184 @@
+
+## Building the WebAssembly component
+
+We can build a WebAssembly component binary out of this JS project with `jco`:
+
+```console
+npm run build
+```
+
+A WebAssembly binary will be written to `string-reverse.wasm`.
+
+## Serving web requests with the WebAssembly component
+
+To run the component and serve requests we can either use `jco` or `wasmtime`:
+
+```console
+$ jco serve http-hello-world.wasm
+Server listening on 8000...
+```
+
+Similarly you can also use `wasmtime`:
+
+```
+$ wasmtime serve -S common http-hello-world.wasm
+Serving HTTP on http://0.0.0.0:8080/
+```
+
+With either approach, you can use `curl` the appropriate URL to trigger your WebAssembly component.
+
+> [!NOTE]
+> The implementations of `jco serve` and `wasmtime serve` are what actually *fulfill* all the imports
+> of your component (see combined/merged `world root` above), and use the `wasi:http/incoming-handler` *export*
+> to make web serving actually happen.
+
+# How it works
+
+## Exploring this Component WIT
+
+As WebAssembly components are powered by a [WebAssembly Interface Types ("WIT")][wit]-first workflow, making
+a HTTP handler component in WebAssembly requires creating a WIT contract to represent that component.
+
+This folder contains a `wit` directory (by convention) with the following content in `component.wit`:
+
+```wit
+package example:http-hello-world;
+
+world component {
+    export wasi:http/incoming-handler@0.2.0;
+}
+```
+
+> [!NOTE]
+> See [`wit/component.wit`](./wit/component.wit)
+>
+> For more information on the WIT syntax, see [WIT design docs][wit]
+
+We make use of the [WebAssembly System Interface ("WASI") HTTP][wasi-http] interface here, pulling in
+pre-established interfaces interfaces for serving *incoming* HTTP requests.
+
+[wasi-http]: https://github.com/WebAssembly/wasi-http
+[wit]: https://github.com/WebAssembly/component-model/blob/main/design/mvp/WIT.md
+
+## Resolving references WebAssembly types
+
+As we intend to use the WASI HTTP interface, we need to pull in WIT interface(s) and types that are referred to by
+the `wasi:http/incoming-handler` interface.
+
+One way fo doing this is *downloading* the WIT from Bytecode Alliance repositories, using [`wkg`, from the `bytecodealliance/wasm-pkg-tools`][wkg].
+
+Since WASI is a growing standard, and well integrated we can generally follow the error messages:
+
+```console
+wkg get wasi:http@0.2.0
+wkg get wasi:random@0.2.0
+wkg get wasi:cli@0.2.0
+wkg get wasi:filesystem@0.2.0
+wkg get wasi:sockets@0.2.0
+wkg get wasi:io@0.2.0
+wkg get wasi:clocks@0.2.0
+```
+
+> [!NOTE]
+> How do we know all these are required? After getting `wasi:http` you can generally follow the error messages.
+
+This will add many WIT files to your local repository, but you can move/rename all the downloaded `*.wit` files
+by making a folder named `deps` under `wit` and dropping them there.
+
+```console
+mkdir wit/deps
+mv *.wit wit/deps
+```
+
+### Typescript support
```

Comment:
I don't think we should be documenting `jco types` for guest component construction since it is not guest component typing compatible, at least until we have an implementation of `jco guest-types` or similar.

Alternatively we could carefully nuance that the types output by `jco types` are the types used when consuming the component and not when authoring it.

Reply:
Yeah let's remove it for now, and when the other PR lands that adds `jco guest-types` we'll use that!